Add beremiz runtime side (beremiz_service.py) package.

TODO: adding host Beremiz.

Signed-off-by: Fabien Marteau <fabien.marteau@armadeus.com>
---

Index: buildroot/package/Config.in
===================================================================
--- buildroot.orig/package/Config.in	2013-07-19 13:36:05.527970007 +0200
+++ buildroot/package/Config.in	2013-07-19 14:29:27.255886081 +0200
@@ -119,6 +119,7 @@
 menu "Graphic libraries and applications (graphic/text)"
 
 comment "Graphic applications"
+source "package/beremiz/Config.in"
 source "package/gnuplot/Config.in"
 source "package/jhead/Config.in"
 source "package/rrdtool/Config.in"
Index: buildroot/package/beremiz/Config.in
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ buildroot/package/beremiz/Config.in	2013-07-19 13:37:01.331968545 +0200
@@ -0,0 +1,19 @@
+comment "Beremiz needs Python 2.7 and pyro"
+	depends on !BR2_PACKAGE_PYTHON || !BR2_PACKAGE_PYTHON_PYRO
+
+config BR2_PACKAGE_BEREMIZ
+	bool "beremiz runtime"
+	depends on BR2_PACKAGE_PYTHON
+	depends on BR2_PACKAGE_PYTHON_PYRO
+	help
+	  This option enable beremiz runtime
+
+	  http://www.beremiz.org/
+
+config BR2_PACKAGE_HOST_BEREMIZ
+	bool "beremiz framework"
+	depends on BR2_PACKAGE_HOST_MATIEC
+	help
+	  This option enable beremiz host framework
+
+	  http://www.beremiz.org/
Index: buildroot/package/beremiz/beremiz.mk
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ buildroot/package/beremiz/beremiz.mk	2013-07-19 14:11:18.343914624 +0200
@@ -0,0 +1,25 @@
+#############################################################
+#
+# Beremiz
+#
+#############################################################
+BEREMIZ_VERSION =358db9d64aa1
+BEREMIZ_SOURCE  = beremiz-$(BEREMIZ_SERVICE_VERSION).tar.gz
+BEREMIZ_SITE    = http://dev.automforge.net/beremiz
+BEREMIZ_SITE_METHOD = hg
+
+BEREMIZ_DEPENDENCIES = python python-pyro
+HOST_BEREMIZ_DEPENDENCIES = host-matiec
+
+define BEREMIZ_INSTALL_TARGET_CMDS
+		mkdir -p $(TARGET_DIR)/opt/beremiz/
+		cp -r $(@D)/* $(TARGET_DIR)/opt/beremiz/
+endef
+
+define HOST_BEREMIZ_INSTALL_CMDS
+		mkdir -p $(HOST_DIR)/opt/beremiz/
+		cp -r $(@D)/* $(HOST_DIR)/opt/beremiz/
+endef
+
+$(eval $(generic-package))
+$(eval $(host-generic-package))
Index: buildroot/package/beremiz/host-beremiz-fix_compare_and_swap_problem_to_be_removed_with_kernel_3.1.patch
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ buildroot/package/beremiz/host-beremiz-fix_compare_and_swap_problem_to_be_removed_with_kernel_3.1.patch	2013-07-19 14:29:40.107885746 +0200
@@ -0,0 +1,30 @@
+Index: beremiz_trunk/targets/Linux/plc_Linux_main.c
+===================================================================
+--- beremiz_trunk.orig/targets/Linux/plc_Linux_main.c	2013-07-19 14:27:11.655889635 +0200
++++ beremiz_trunk/targets/Linux/plc_Linux_main.c	2013-07-19 14:27:54.111888523 +0200
+@@ -17,10 +17,25 @@
+ {
+     return __sync_val_compare_and_swap(atomicvar, compared, exchange);
+ }
++/*
+ long long AtomicCompareExchange64(long long* atomicvar, long long compared, long long exchange)
+ {
+     return __sync_val_compare_and_swap(atomicvar, compared, exchange);
+ }
++*/
++static pthread_mutex_t cs_mutex = PTHREAD_MUTEX_INITIALIZER;
++long long AtomicCompareExchange64(long long* atomicvar, long long compared, long long exchange)
++{
++    long long res;
++ 
++    pthread_mutex_lock(&cs_mutex);
++    res=*atomicvar;
++    if(*atomicvar == compared){
++        *atomicvar = exchange;
++    }
++    pthread_mutex_unlock(&cs_mutex);
++    return res;
++}
+ 
+ void PLC_GetTime(IEC_TIME *CURRENT_TIME)
+ {
